src/lib/config-utils.ts

```diff
@@ -0,0 +1,376 @@
+	} else {
+		throw new Error(`No boot config var regex for device type: ${deviceType}`);
+	}
+}
+
+export function envToBootConfig(deviceType: string, env: EnvVarObject): ConfigOptions {
+	if (!isConfigDeviceType(deviceType)) {
+		return { };
+	}
+	const prefix = bootConfigVarPrefix(deviceType);
+	const regex = bootConfigVarRegex(deviceType);
+
+	let parsedEnv = _.pickBy(env, (_val: string, key: string) => {
+		return _.startsWith(key, prefix);
+	});
+	parsedEnv = _.mapKeys(parsedEnv, (_val, key) => {
+		return key.replace(regex, '$2');
+	});
+	parsedEnv = _.mapValues(parsedEnv, (val, key) => {
+		if (isRPiDeviceType(deviceType)) {
+			if (_.includes(rpiArrayConfigKeys, key)) {
+				if (!_.startsWith(val, '"')) {
+					return [ val ];
+				} else {
+					return JSON.parse(`[${val}]`);
+				}
+			}
+		}
+		return val;
+	});
+
+	return parsedEnv as ConfigOptions;
+}
+
+export function bootConfigToEnv(deviceType: string, config: ConfigOptions): EnvVarObject {
+	const prefix = bootConfigVarPrefix(deviceType);
+	const confWithEnvKeys = _.mapKeys(config, (_val, key) => {
+		return prefix + key;
+	});
+	return _.mapValues(confWithEnvKeys, (val) => {
+		if (_.isArray(val)) {
+			return JSON.stringify(val).replace(/^\[(.*)\]$/, '$1');
+		}
+		return val;
+	});
+}
+
+export function filterConfigKeys(
+	deviceType: string,
+	allowedKeys: string[],
+	conf: { [key: string]: any },
+): { [key: string]: any } {
+
+	let isConfigType: boolean = false;
+	let prefix: string;
+	if (isConfigDeviceType(deviceType)) {
+		prefix = bootConfigVarPrefix(deviceType);
+		isConfigType = true;
+	}
+	return _.pickBy(conf, (_v, k) => {
+		return _.includes(allowedKeys, k) || (isConfigType && _.startsWith(k, prefix));
+	});
+}
+
+export function getBootConfigPath(deviceType: string): string {
+	if (isRPiDeviceType(deviceType)) {
+		return getRpiBootConfig();
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return getExtlinuxBootConfig();
+	} else {
+		throw new Error(`No boot config exists for device type: ${deviceType}`);
+	}
+}
+
+function getRpiBootConfig(): string {
+	return bootMountPoint + '/config.txt';
+}
+
+function getExtlinuxBootConfig(): string {
+	return bootMountPoint + '/extlinux/extlinux.conf';
+}
+
+export function parseBootConfig(deviceType: string, conf: string): ConfigOptions {
+	if (isRPiDeviceType(deviceType)) {
+		return parseRpiBootConfig(conf);
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return parseExtlinuxBootConfig(conf);
+	} else {
+		throw new Error(`Cannot parse boot config for device type: ${deviceType}`);
+	}
+}
+
+export function parseRpiBootConfig(confStr: string): ConfigOptions {
+	const conf: ConfigOptions = { };
+	const configStatements = confStr.split(/\r?\n/);
+
+	for (const configStr of configStatements) {
+		// Don't show warnings for comments and empty lines
+		const trimmed = _.trimStart(configStr);
+		if (_.startsWith(trimmed, '#') || trimmed === '') {
+			continue;
+		}
+		let keyValue = /^([^#=]+)=(.+)/.exec(configStr);
```

Comment:
This regex is slightly better I think: `/^([^=]+)=(.*)$/`. Essentially, it matches everything until a `=` as the key and everything until EOL as the value and also allows a value to be empty.